File: remote-ickle-query/src/main/scala/org/littlewings/infinispan/icklequery/Book.scala
package org.littlewings.infinispan.icklequery

import org.infinispan.protostream.annotations.{ProtoField, ProtoMessage}
import org.infinispan.protostream.descriptors.Type

import scala.annotation.meta.beanGetter
import scala.beans.BeanProperty

object Book {
  def apply(isbn: String, title: String, price: Int, category: String): Book = {
    val book = new Book
    book.isbn = isbn
    book.title = title
    book.price = price
    book.category = category
    book
  }
}

@ProtoMessage(name = "Book")
class Book {
  @BeanProperty
  @(ProtoField @beanGetter)(number = 1, name = "isbn", `type` = Type.STRING)
  var isbn: String = _

  @BeanProperty
  @(ProtoField @beanGetter)(number = 2, name = "title", `type` = Type.STRING)
  var title: String = _

  @BeanProperty
  @(ProtoField @beanGetter)(number = 3, name = "price", `type` = Type.INT32, defaultValue = "0")
  var price: Int = _

  @BeanProperty
  @(ProtoField @beanGetter)(number = 4, name = "category", `type` = Type.STRING)
  var category: String = _
}

File: remote-new-hotrod-client-underconstruction/src/main/java/org/littlewings/infinispan/remote/newclient/Book.java
package org.littlewings.infinispan.remote.newclient;

import org.infinispan.protostream.annotations.ProtoFactory;
import org.infinispan.protostream.annotations.ProtoField;
import org.infinispan.protostream.descriptors.Type;

public class Book {
    @ProtoField(number = 1)
    String isbn;

    @ProtoField(number = 2)
    String title;

    @ProtoField(number = 3, type = Type.INT32, defaultValue = "0")
    int price;

    @ProtoFactory
    public static Book create(String isbn, String title, int price) {
        Book book = new Book();
        book.setIsbn(isbn);
        book.setTitle(title);
        book.setPrice(price);

        return book;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }
}
